src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -4176,7 +4175,7 @@ static IDisposable OpenReadTransaction(QueryOperationContext context)
         {
             var dict = new Dictionary<string, HashSet<string>>(StringComparer.OrdinalIgnoreCase);
             if (Status == IndexRunningStatus.Disabled || Status == IndexRunningStatus.Paused)
-                dict[Name] = Collections;
+                dict[TombstoneCleanerIdentifier] = Collections;
```

Comment:
something is off, in one place we are using this, is other we are creating new names. We need to normalize this and use the same mechanism everywhere IMO

Reply:
I'm not sure if I overdid it with the normalization, but I implemented a new interface called `ITombstoneDeletionBlocker`, which is implemented by anyone who can prevent the deletion of `Tombstones`. This interface includes a read-only property with a clear name for the source of the block and a `Disabled` property to determine if this less obvious block is being applied. I was able to implement this interface everywhere except the Index class, as it itself is an implementer of the `ITombstoneAware` interface...

By the way, I didn't quite understand why it was done this way... At a glance, it seems to me it would be more logical to make `IndexStore` implement the `ITombstoneAware` interface? Or maybe I'm misunderstanding something...